mandible/umm/ummg.py

```diff
@@ -0,0 +1,221 @@
+from dataclasses import dataclass
+from datetime import datetime, timezone
+from pathlib import Path
+from typing import List, TypedDict
+
+# ISO 8601 with Z
+UMM_DATETIME_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"
+
+
+def _remove_missing(ummg: dict) -> dict:
+    return {k: v for k, v in ummg.items() if v}
+
+
+class FileMd(TypedDict):
+    name: str
+    uri: str
+    s3uri: str
+    size: int
+    key: str
+    bucket: str
+
+
+class CollectionRef(TypedDict, total=False):
+    short_name: str
+    long_name: str
+
+
+class ProductMd(TypedDict, total=False):
+    product_creation_datetime: datetime
+    product_start_time: datetime
+    product_stop_time: datetime
+    mission: str
+
+
+@dataclass
+class ProductFilesMd:
+    productFile: FileMd
+    relatedFiles: List[FileMd]
+
+
+@dataclass
+class Meta:
+    collection_ref: CollectionRef
+    product_md: ProductMd
+    product_files_md: ProductFilesMd
```

Comment:
Make them all the same. Right now there's a mix of typeddict and dataclass

Reply:
done!